src/Slider/Slider.js

```diff
@@ -0,0 +1,263 @@
+// @flow
+
+import * as React from 'react';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import { View } from 'react-native';
+import MultiSlider from 'react-native-multi-slider';
+
+import Marker from './Marker';
+import { Text } from '../Text';
+import { StyleSheet } from '../PlatformStyleSheet';
+
+const LABEL_LENGTH = 200;
+const TRACK_HEIGHT = 4;
+
+type Props = {|
+  +type: 'multi' | 'single',
+  +minValue: number,
+  +maxValue: number,
+  +startValue: number,
+  +endValue?: number,
+  +label: string | React.Node,
+  +numOfParts?: number,
+  +snapped?: boolean,
+  +step?: number,
+  +customMarker?: React.Node,
+|};
+
+type OnLayout = {
+  +nativeEvent: {
+    +layout: {
+      +x: number,
+      +y: number,
+      +width: number,
+      +height: number,
+    },
+  },
+};
+
+type State = {
+  multiSliderValues: Array<number>,
+  singleSliderValue: Array<number>,
+  width: number,
+};
+
+export default class Slider extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    const { maxValue, startValue, endValue = maxValue } = this.props;
+
+    if (endValue > maxValue) {
+      console.error('End value cannot be equal or higher than maximum value');
+    }
+
+    this.state = {
+      width: 0,
+      multiSliderValues: [startValue, endValue],
+      singleSliderValue: [startValue],
+    };
+  }
+
+  sliderOneValuesChange = (values: Array<number>) => {
+    this.setState({
+      singleSliderValue: values,
+    });
+  };
+
+  multiSliderValuesChange = (values: Array<number>) => {
+    this.setState({
+      multiSliderValues: values,
+    });
+  };
+
+  calculateOffset = (value: number) => {
+    const { minValue, maxValue } = this.props;
+    const { width } = this.state;
+    return (value * width) / (maxValue - minValue) - LABEL_LENGTH / 2;
+  };
+
+  createParts = () => {
```

Comment:
What about extracting this part into some separate `TrackLine` component? Maybe even with those current two `Views` with `trackLine` style.